lib/elixir/lib/access.ex

```diff
@@ -119,14 +83,67 @@ defmodule Access do
         languages: [%{name: "ELIXIR", type: :functional},
                     %{name: "C", type: :procedural}]}
 
-  See the functions `key/1`, `key!/1`, `elem/1`, and `all/0` for some of the
-  available accessors.
+  See the functions `at/1`, `key/1`, `key!/1`, `elem/1`, and `all/0` for some of the
+  available helpers.
+
+  ## Structs and Dot Syntax
+
+  As described earlier, the bracket syntax is disabled by default for struct types.
+  However, Elixir provides a "dot syntax" which can be used to traverse structs and
+  maps (with atoms for keys). It differs from bracket syntax because _it will raise for
+  non-existing keys_. This is to emphasize its usage for data structures with a _static_
+  set of keys (like struct types) whereas bracket syntax is designed for
+  data-structures with a _dynamic_ set of keys.
+
+      user = %User{name: "John"}
+      user.name
+      #=> "John"
+      
+      user.not_a_key
+      ** (KeyError) key :not_a_key not found in: %User{name: "John"}
+
+  Dot syntax does not rely on the Access behavior and as such is restricted to structs
+  and atom keys in maps.
+
+  ## The Kernel Accessors Can Use Bracket and Dot Syntax
+  
+  `Kernel` also offers supplemental macros matching the accessor functions. The macros
+  interpret both bracket and dot syntax before passing the keys on to the matching
+  function.
+  
+  For example, you may convert:
+  
+      iex> map = %{a: 1, b: 2}
+      iex> put_in map, [:a], 3
+      %{a: 3, b: 2}
+  
+  to:
+  
+      iex> map = %{a: 1, b: 2}
+      iex> put_in map[:a], 4
+      %{a: 4, b: 2}
+
+  This syntax can be nested arbitrarily:
+
+      iex> users = %{"john" => %{age: 27}, "meg" => %{age: 23}}
+      iex> put_in users["john"][:age], 28
+      %{"john" => %{age: 28}, "meg" => %{age: 23}}
+
+  Using dot syntax follows the rules of raising for non existing keys.
+  
+      iex> map = %{a: 1, b: 2}
+      iex> put_in map.a, 5
+      %{a: 5, b: 2}
+      iex> put_in map.c, 5
+      ** (KeyError) key :c not found in: %{a: 1, b: 2}
+
+  ## Implementing the Behaviour
 
-  ## Implementing the Access behaviour for custom data structures
+  Since Access defines a behaviour, it can be implemented for arbitrary key-value data
```

Comment:
The only reason you would do so is if the struct actually implements a key-value data structure.